### PR TITLE
UHM-4979 OVC Intake form pt 2

### PR DESCRIPTION
**Pull request by brandones**
Configure OVC program; don't show OVC forms for patient older than 21

Related to https://github.com/PIH/openmrs-config-pihemr/pull/88

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java [hunk 1]
```diff
@@ -112,20 +113,7 @@
...
-import static org.openmrs.module.mirebalais.require.RequireUtil.patientIsAdult;
-import static org.openmrs.module.mirebalais.require.RequireUtil.patientIsChild;
-import static org.openmrs.module.mirebalais.require.RequireUtil.patientIsFemale;
-import static org.openmrs.module.mirebalais.require.RequireUtil.patientNotDead;
-import static org.openmrs.module.mirebalais.require.RequireUtil.patientVisitWithinPastThirtyDays;
-import static org.openmrs.module.mirebalais.require.RequireUtil.sessionLocationHasTag;
-import static org.openmrs.module.mirebalais.require.RequireUtil.userHasPrivilege;
-import static org.openmrs.module.mirebalais.require.RequireUtil.visitDoesNotHaveEncounterOfType;
-import static org.openmrs.module.mirebalais.require.RequireUtil.visitHasEncounterOfType;
+import static org.openmrs.module.mirebalais.require.RequireUtil.*;
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java:116]: Did you deliberately switch to wildcard imports?  As an OpenMRS convention, we don't use wildcard imports (though I'm not totally opposed to it in this case).  IntelliJ can be set to not substitute wilcards.

File: api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java [hunk 2]
```diff
@@ -2199,21 +2183,25 @@ private void enableCovid19() {
...
+
         extensions.add(visitAction(Extensions.OVC_INITIAL_VISIT_ACTION,
                 "ui.i18n.EncounterType.name." + EncounterTypes.OVC_INTAKE.uuid(),
                 "fas fa-fw fa-child",
                 "link",
                 enterStandardHtmlFormLink(PihCoreUtil.getFormResource("ovcIntake.xml")),
                 null,
-                visitDoesNotHaveEncounterOfType(EncounterTypes.OVC_INTAKE)));
+                and(or(patientAgeUnknown(), patientYoungerThan(22)),
+                        visitDoesNotHaveEncounterOfType(EncounterTypes.OVC_INTAKE))));
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java:2195]: For HIV Intake and Followup I think we don't allow an Intake if there is either a Intake or a Followup on the same visit... (but maybe that's not the same requirement here, I haven't checked, but wanted to throw it out there).

**Review by mogoodrich — COMMENTED** [review]
Make a couple comments, but generally LGTM...

File: api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java (hunk 1 above)
Comment (brandones) [reply api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java:116]: No, I just need to configure IntelliJ correctly. Sorry about that.

File: api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java (hunk 2 above)
Comment (brandones) [reply api/src/main/java/org/openmrs/module/mirebalais/apploader/CustomAppLoaderFactory.java]: That's sensible enough. The requirement is actually "no followup until there's an intake, no intake after there's already an intake," but we don't support those kinds of conditions (i.e., conditions across all visits/encounters).

**Pull request merged**